### ISSUE TITLE
Import cover image by aspect ratio

### ISSUE THREAD

**Issue by tomblind** [JosefNemec/Playnite#1888]
To help with consistency in the Grid View, it would be great if the cover image import had an option to select the image from all sources that comes closest to the target aspect ratio. I imagine there would be an option to prefer wide or tall.

If at some point #1355 is supported (I imagine it would have to be per-filter aspect ratios using #590), it might even be worth storing multiple cover images so the best one could be dynamically selected based on the current view.

**Comment by JosefNemec** [comment]
This is not possible. Playnite has no idea what the media file properties are, it's all handled by the metadata plugin(s), and those could in theory provide covers in different aspect ratios anyways (there's no standard).

You can already change priorities of metadata sources in application settings to prioritize the one that fits your settings. Just select the source that provides images in aspect ratio that you want.

**Comment by tomblind** [comment]
Yes, this is what I do now, but there's plenty of inconsistent results and I spend a lot of time manually correcting them. IGDB is the most consistent but still gives some 'banner' style images instead of the portrait style it normally does. Also it often gets images for the wrong game entirely. I'd prefer to use the officail store image IF it's close to my target ratio, and IGDB otherwise.

If Playnite actually stored the images from all sources, it would be able to see and compare the sizes though, right?

Alternatively, would there be a way do do this via extension? Would one extension be able to use the others and do the final logic?

**Comment by JosefNemec** [comment]
That would require Playnite to pull images from all available sources, compare them and then use one. That would be incredibly slow.

No, extensions can't call other extensions, you would have to implement functionality from those extensions into your custom one.
